esp32c2-hal/examples/interrupt_preemption.rs

```diff
@@ -0,0 +1,147 @@
+//! Software Interrupts
+//!
+//! An example of how software interrupts can be raised and reset
+//! Should rotate through all of the available interrupts printing their number
+//! when raised.
```

Comment:
Looks like this & the other two examples need a new doc comment. I gave it a try, take a look and see what you think?

```suggestion
//! Interrupt Preemption
//!
//! An example of how an interrupt can be preempted by another with higher priority.
//! Should show higher-numbered software interrupts happening during the handling of lower-numbered ones.
```

Reply:
LGTM, forgot to modify the source examples, thanks!